OpenMined/syft
# App Install Small Fix

## Description

Main Issue :  The app.json file isn't generated correctly when more than one job is installed.
- Local app installation: We are not currently including locally installed apps in the app.json file.
- New path field: A path field has been added for each app in the app.json file.
- This will enable us to use the app.json file to list and uninstall apps in the future.
- Fix flaky timeout approach for checking if a app repository is valid or not.

## Affected Dependencies
- syftbox/app/install.py
- syftbox/app/manager.py

## How to test it
- Install two or more remote apps with `config.json` in it. It must generate a `apps.json` in the following structure.
```
{
    "OpenMined/fl_ring": {
        "commit": "d2c19e09d1417fa6b8a763825ef8f96056fae5fd",
        "version": "0.1.0",
        "path": "/path/to/user/sync/apps/fl_ring",
    },
    "OpenMined/fl_ring_dash": {
        "commit": "02e58f9b2227d2434cde540253f43ba870f0a74a",
        "version": "0.1.0",
        "path": "/path/to/user/sync/apps/fl_ring_dash",
    },
}
```
- Execute `syftbox app install InvalidUser/InvalidRepository` and check if it fails properly
